### DISCUSSION TITLE
Starting Display on top left instead of center canvas/paper

### DISCUSSION THREAD

**Discussion by pandika-csgi** [clientIO/joint#2188] [Q&A] [answered]
### Introduction

When have wide paper or huge paper, I want starting point for display to be on top left corner or left center instead of top center of paper

### Case 1
![image](https://github.com/clientIO/joint/assets/106044262/494543f3-d21a-489b-a300-92359c441671)

### Expected: 
![image](https://github.com/clientIO/joint/assets/106044262/3ce417de-e55e-4a8b-b37f-f80e3069e432)

### Case 2
![image](https://github.com/clientIO/joint/assets/106044262/cc7e48b9-7d1c-420e-8e96-1dd918769aeb)

### Expected:
![image](https://github.com/clientIO/joint/assets/106044262/0cde8cdd-eb94-4ee2-8d06-acbe9a3916f1)

How to do that? I've read the documentation https://resources.jointjs.com/docs/rappid/v3.6/ui.PaperScroller.html, but dont know what the number to put inside 

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
You can use the `positionPoint()` method of PaperScroller. Use this to tell the paper scroller to place a point in the graph coordinate system at any side of the paper scroller container.

```js
paperScroller.positionPoint(paper.getArea().topLeft().offset(-20, -20), 'top-left')
```

**Reply by pandika-csgi** [reply]
Thanks for your response @kumilingus, I've tried that, but got an error because in `rappid.d.ts`  for `positionPoint` is like this `positionPoint(point: g.Point, x: number, y: number, opt?: { [key: string]: any }): this;` for `x` and `y` need to pass `number` can't be `string`

I've also tried this 
```ts
paperScroller.positionPoint(paper.getArea().topLeft().offset(-20, -20), 0, 0);
```
It's working on a new paper, my initial paper was `{width: 1000, height: 600}`
because Im loaded an existing diagram and the paper was expanded, how to get that loaded paper size, im always got my initial paper `{width: 1000, height: 600}`

**Reply by kumilingus** [reply, marked as answer]
My bad. You are right. It's `paperScroller.positionPoint(app.paper.getArea().topLeft().offset(-20, -20), 0, 0);` what I meant.
It works as expected for me (even if the paper gets expanded). Could you please elaborate more?

If you are interested in the size of the paper then you can use:
1. `paper.getComputedSize()` - this will return the size of the paper in client coordinates i.e. the actual size on the screen (if the paper is zoomed, the size is larger)
2. `paper.getArea()` - the area of the paper it covers in graph coordinates i.e. `area.x` is the leftmost point, `area.y` is the uppermost point, `area.x + area.width` is the rightmost point, `area.y + area.height` is the bottommost point.

*Note: the TypeScript definition should be: `positionPoint(point: g.Point, x: number | string, y: number | string, opt?: { [key: string]: any }): this;`*

**Reply by pandika-csgi** [reply]
Thanks, for the TypeScript definition, which may be different because the version I used is different, but all good, I finally use this `paperScroller.positionContent('top-left', { padding: { horizontal: 100, vertical: 100 }, useModelGeometry: true })`, if I'm not wrong that based on element right?

**Reply by kumilingus** [reply]
Right, it works differently. The `positionPoint()` is more generic.
### 1. positionPoint()
```js
paperScroller.positionPoint(paper.getArea().topLeft(), 0, 0);
```
<img width="333" alt="image" src="https://github.com/clientIO/joint/assets/3967880/e98d439b-51b2-4211-a2ce-84ebd31bc40e">

### 2. positionContent()
```js
paperScroller.positionContent('top-left', { useModelGeometry: true });
```

<img width="341" alt="image" src="https://github.com/clientIO/joint/assets/3967880/2090d74b-d607-42d4-8540-eecce6af4085">

which equals to:
```js
paperScroller.positionPoint(graph.getBBox().topLeft(), 0, 0);
```
